In [1]:
import pandas as pd

Loading Data

In [3]:
#df = pd.read_csv(r'‪C:\Users\tlynx\Desktop\CBC_Data\Coffee_shop_sale.csv')

df = pd.read_csv(r'C:/Users/tlynx/Desktop/CBC_Data/Coffee_shop_sale.csv')
df

transaction_id        date   time product_category           product_name  \
0                1  2024-03-01  08:15           Coffee               Espresso   
1                2  2024-03-01  08:17           Pastry              Croissant   
2                3  2024-03-01  08:45           Coffee             Cappuccino   
3                4  2024-03-01  09:30              Tea              Green Tea   
4                5  2024-03-01  10:05           Coffee                  Latte   
..             ...         ...    ...              ...                    ...   
95              96  2024-03-10  14:40         Smoothie         Tropical Green   
96              97  2024-03-10  16:00           Pastry  Chocolate Chip Cookie   
97              98  2024-03-10  17:15           Coffee              Americano   
98              99  2024-03-10  18:05              Tea             Herbal Tea   
99             100  2024-03-10  18:45           Coffee           Decaf Coffee   

       size  price  quantity  transaction_total payment_method  
0     Small   2.50         1               2.50           Cash  
1   Regular   3.25         1               3.25    Credit Card  
2    Medium   4.75         1               4.75     Debit Card  
3    Medium   3.50         2               7.00           Cash  
4     Large   5.25         1               5.25    Credit Card  
..      ...    ...       ...                ...            ...  
95  Regular   5.75         1               5.75           Cash  
96  Regular   2.50         1               2.50     Debit Card  
97    Small   2.75         1               2.75    Credit Card  
98   Medium   3.25         1               3.25           Cash  
99   Medium   2.75         1               2.75     Debit Card  

[100 rows x 10 columns]

INSPECTING YOUR DATA

In [14]:
#Viewing you data
df.head(5)

transaction_id        date   time product_category product_name     size  \
0               1  2024-03-01  08:15           Coffee     Espresso    Small   
1               2  2024-03-01  08:17           Pastry    Croissant  Regular   
2               3  2024-03-01  08:45           Coffee   Cappuccino   Medium   
3               4  2024-03-01  09:30              Tea    Green Tea   Medium   
4               5  2024-03-01  10:05           Coffee        Latte    Large   

   price  quantity  transaction_total payment_method  
0   2.50         1               2.50           Cash  
1   3.25         1               3.25    Credit Card  
2   4.75         1               4.75     Debit Card  
3   3.50         2               7.00           Cash  
4   5.25         1               5.25    Credit Card

In [15]:
df.tail(3)

transaction_id        date   time product_category  product_name    size  \
97              98  2024-03-10  17:15           Coffee     Americano   Small   
98              99  2024-03-10  18:05              Tea    Herbal Tea  Medium   
99             100  2024-03-10  18:45           Coffee  Decaf Coffee  Medium   

    price  quantity  transaction_total payment_method  
97   2.75         1               2.75    Credit Card  
98   3.25         1               3.25           Cash  
99   2.75         1               2.75     Debit Card

In [4]:
#it picks a random sample
df.sample(6)


transaction_id        date   time product_category      product_name  \
4                5  2024-03-01  10:05           Coffee             Latte   
77              78  2024-03-08  16:35           Coffee      Black Coffee   
31              32  2024-03-04  09:10           Pastry  Blueberry Muffin   
18              19  2024-03-02  16:30              Tea         Earl Grey   
82              83  2024-03-09  10:45           Coffee        Cappuccino   
85              86  2024-03-09  14:30         Smoothie       Berry Blast   

       size  price  quantity  transaction_total payment_method  
4     Large   5.25         1               5.25    Credit Card  
77    Large   3.00         1               3.00           Cash  
31  Regular   3.50         1               3.50    Credit Card  
18    Large   4.25         1               4.25     Debit Card  
82   Medium   4.75         1               4.75    Credit Card  
85  Regular   5.75         1               5.75    Credit Card

In [8]:
#summary information
df.info()
df

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   transaction_id     100 non-null    int64  
 1   date               100 non-null    object 
 2   time               100 non-null    object 
 3   product_category   100 non-null    object 
 4   product_name       100 non-null    object 
 5   size               100 non-null    object 
 6   price              100 non-null    float64
 7   quantity           100 non-null    int64  
 8   transaction_total  100 non-null    float64
 9   payment_method     100 non-null    object 
dtypes: float64(2), int64(2), object(6)
memory usage: 7.9+ KB


transaction_id        date   time product_category           product_name  \
0                1  2024-03-01  08:15           Coffee               Espresso   
1                2  2024-03-01  08:17           Pastry              Croissant   
2                3  2024-03-01  08:45           Coffee             Cappuccino   
3                4  2024-03-01  09:30              Tea              Green Tea   
4                5  2024-03-01  10:05           Coffee                  Latte   
..             ...         ...    ...              ...                    ...   
95              96  2024-03-10  14:40         Smoothie         Tropical Green   
96              97  2024-03-10  16:00           Pastry  Chocolate Chip Cookie   
97              98  2024-03-10  17:15           Coffee              Americano   
98              99  2024-03-10  18:05              Tea             Herbal Tea   
99             100  2024-03-10  18:45           Coffee           Decaf Coffee   

       size  price  quantity  transaction_total payment_method  
0     Small   2.50         1               2.50           Cash  
1   Regular   3.25         1               3.25    Credit Card  
2    Medium   4.75         1               4.75     Debit Card  
3    Medium   3.50         2               7.00           Cash  
4     Large   5.25         1               5.25    Credit Card  
..      ...    ...       ...                ...            ...  
95  Regular   5.75         1               5.75           Cash  
96  Regular   2.50         1               2.50     Debit Card  
97    Small   2.75         1               2.75    Credit Card  
98   Medium   3.25         1               3.25           Cash  
99   Medium   2.75         1               2.75     Debit Card  

[100 rows x 10 columns]

In [22]:
df.describe()
#it's the statistics summary

transaction_id       price    quantity  transaction_total
count      100.000000  100.000000  100.000000         100.000000
mean        50.500000    3.869000    1.140000           4.281500
std         29.011492    1.092393    0.426875           1.459013
min          1.000000    2.000000    1.000000           2.000000
25%         25.750000    3.000000    1.000000           3.250000
50%         50.500000    3.750000    1.000000           4.000000
75%         75.250000    4.750000    1.000000           5.062500
max        100.000000    5.950000    3.000000           9.500000

In [24]:
df.shape
#it's the length and breadth of your data i.e the rows and columns

(100, 10)

In [25]:
df.dtypes

transaction_id         int64
date                  object
time                  object
product_category      object
product_name          object
size                  object
price                float64
quantity               int64
transaction_total    float64
payment_method        object
dtype: object

In [26]:
#Use .info() to answer:
    #How many rows and columns are there?
    #What are the data types of each colum?

#Summary Stats: Use .describe() on the numerical columns(price, quantity, transaction_total).
    #What is the average transaction total?
    #The most expensive simple item? The Largest quantity sold in one transaction?


df.info()

Columns: 10
Rows: 100
dtypes: float64(2), int64(2), object(6)

In [8]:
df.describe()

transaction_id       price    quantity  transaction_total
count      100.000000  100.000000  100.000000         100.000000
mean        50.500000    3.869000    1.140000           4.281500
std         29.011492    1.092393    0.426875           1.459013
min          1.000000    2.000000    1.000000           2.000000
25%         25.750000    3.000000    1.000000           3.250000
50%         50.500000    3.750000    1.000000           4.000000
75%         75.250000    4.750000    1.000000           5.062500
max        100.000000    5.950000    3.000000           9.500000

Average transaction total: 4.281500
The most expensive single item: 5.950000
The largest quantity sold in one transaction: 3.000000

In [11]:
#categorical summary
df['product_category'].value_counts()
#it counts each unique product

product_category
Coffee      50
Pastry      20
Tea         20
Smoothie    10
Name: count, dtype: int64

In [12]:
#Using the .value_counts() in the payment, size and product name
#1. what is the most used and how many are they?
#2. what is the least and how many are they?

In [17]:
df['payment_method'].value_counts()

payment_method
Cash           34
Credit Card    33
Debit Card     33
Name: count, dtype: int64

Most Used Payment Method: cash 34
How many: 1

Least Used Payment Method:Debit card and credit card 33 
How Many?: 2 2

In [20]:
df['size'].value_counts()

size
Regular    30
Medium     27
Small      24
Large      19
Name: count, dtype: int64

Most Used Size:Regular
How Many Size: 30

Least Used: large 
How Many:19

In [22]:
df['product_name'].value_counts()

product_name
Latte                    10
Espresso                  7
Black Coffee              7
Croissant                 6
Cappuccino                5
Green Tea                 5
Blueberry Muffin          4
Chai Latte                4
Americano                 4
Chocolate Chip Cookie     4
Mocha                     4
Decaf Coffee              4
Banana Bread              3
Herbal Tea                3
Flat White                3
Macchiato                 3
Peppermint Tea            3
Earl Grey                 3
Tropical Green            3
Cinnamon Roll             3
Berry Blast               3
Cold Brew                 3
Vanilla Almond            2
Chamomile Tea             2
Chocolate Banana          2
Name: count, dtype: int64

Most Used Product_Name: Latte
How Many: 10

Least Used Product_Name:Vanilla Almond,Chamomile Tea and Chocolate Banana
How Many:2 2 2